autogalaxy/analysis/analysis.py

```diff
@@ -0,0 +1,483 @@
+    def __init__(
+        self,
+        dataset,
+        hyper_result=None,
+        cosmology=cosmo.Planck15,
+        settings_pixelization=pix.SettingsPixelization(),
+        settings_inversion=inv.SettingsInversion(),
+        preloads=pload.Preloads(),
+        use_result_as_hyper_dataset=False,
+    ):
+
+        super().__init__(hyper_result=hyper_result, cosmology=cosmology)
+
+        self.dataset = dataset
+
+        if self.hyper_result is not None:
+
+            self.set_hyper_dataset(result=self.hyper_result)
+
+        else:
+
+            self.hyper_galaxy_image_path_dict = None
+            self.hyper_model_image = None
+
+        self.settings_pixelization = settings_pixelization
+        self.settings_inversion = settings_inversion
+        self.use_result_as_hyper_dataset = use_result_as_hyper_dataset
+        self.preloads = preloads
+
+    def set_hyper_dataset(self, result):
+
+        self.hyper_galaxy_image_path_dict = result.hyper_galaxy_image_path_dict
+        self.hyper_model_image = result.hyper_model_image
+
+    def hyper_image_sky_for_instance(self, instance):
+
+        if hasattr(instance, "hyper_image_sky"):
+            return instance.hyper_image_sky
+
+    def hyper_background_noise_for_instance(self, instance):
+
+        if hasattr(instance, "hyper_background_noise"):
+            return instance.hyper_background_noise
+
+    def plane_for_instance(self, instance):
+        return pl.Plane(galaxies=instance.galaxies)
+
+    def associate_hyper_images(self, instance: af.ModelInstance) -> af.ModelInstance:
+        """
+        Takes images from the last result, if there is one, and associates them with galaxies in this search
+        where full-path galaxy names match.
+
+        If the galaxy collection has a different name then an association is not made.
+
+        e.g.
+        galaxies.lens will match with:
+            galaxies.lens
+        but not with:
+            galaxies.lens
+            galaxies.source
+
+        Parameters
+        ----------
+        instance
+            A model instance with 0 or more galaxies in its tree
+
+        Returns
+        -------
+        instance
+           The input instance with images associated with galaxies where possible.
+        """
+
+        if self.hyper_galaxy_image_path_dict is not None:
+
+            for galaxy_path, galaxy in instance.path_instance_tuples_for_class(
+                g.Galaxy
+            ):
+                if galaxy_path in self.hyper_galaxy_image_path_dict:
+                    galaxy.hyper_model_image = self.hyper_model_image
+
+                    galaxy.hyper_galaxy_image = self.hyper_galaxy_image_path_dict[
+                        galaxy_path
+                    ]
+
+        return instance
+
+    def make_attributes(self):
+        raise NotImplementedError
+
+    def save_settings(self, paths: af.Paths):
+
+        with open(f"{paths.pickle_path}/settings_masked_dataset.pickle", "wb+") as f:

```

Comment:
I think these methods will have to return a dictionary of settings rather than implementing saving them. That will make the code more concise too.